app/controllers/hosts_controller.rb

```diff
@@ -47,8 +47,13 @@ def index(title = nil)
       format.html do
         @hosts = search.includes(included_associations).paginate(:page => params[:page], :per_page => params[:per_page])
         # SQL optimizations queries
-        @last_report_ids = ConfigReport.where(:host_id => @hosts.map(&:id)).group(:host_id).maximum(:id)
-        @last_reports = ConfigReport.where(:id => @last_report_ids.values)
+        if defined? ForemanHostReports
+          @last_report_ids = HostReport.where(:host_id => @hosts.map(&:id)).reorder('').group(:host_id).maximum(:id)
+          @last_reports = HostReport.where(:id => @last_report_ids.values)
+        else
+          @last_report_ids = ConfigReport.where(:host_id => @hosts.map(&:id)).reorder('').group(:host_id).maximum(:id)
+          @last_reports = ConfigReport.where(:id => @last_report_ids.values)
+        end
```

Comment:
This would end up being ugly hack in the plugin, it is just a SQL optimalization.

Reply:
I can probably do this via filter. Let me look on this.

Reply:
Actually not, an explicit method is needed.